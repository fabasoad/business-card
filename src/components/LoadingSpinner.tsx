import * as React from 'react'

export default function LoadingSpinner() {
  return (
    <div className="loader">
      <div className="lds-spinner">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
    </div>
  )
}
