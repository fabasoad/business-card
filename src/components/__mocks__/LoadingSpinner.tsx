import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
