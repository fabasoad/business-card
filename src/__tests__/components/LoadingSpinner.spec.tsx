/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import LoadingSpinner from '../../components/LoadingSpinner'

test('should render LoadingSpinner correctly', () => {
  const wrapper = shallow(<LoadingSpinner />)
  expect(wrapper).toMatchSnapshot()
})
