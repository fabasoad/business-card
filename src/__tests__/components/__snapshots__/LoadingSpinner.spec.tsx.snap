// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoadingSpinner correctly 1`] = `
<div
  className="loader"
>
  <div
    className="lds-spinner"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
